src/type.ml

```diff
@@ -5,6 +5,7 @@ open Monad.Notations
 
 type t =
   | Variable of Name.t
+  | SetTyp
```

Comment:
I guess this is to talk about the "kind" of the types, either a polymorphic variable in `Set` or a tag. Can this be a separate type for "kinds"?